src/membrane_proof.rs

```diff
@@ -87,52 +83,61 @@ fn mem_proof_server_url() -> String {
     }
 }
 
-pub fn get_hpos_config() -> Result<Config> {
-    let config_path = env::var("HPOS_CONFIG_PATH")?;
-    let config_json = fs::read(config_path)?;
-    let config: Config = serde_json::from_slice(&config_json)?;
-    Ok(config)
+/// Returns memproof from a file at MEM_PROOF_PATH
+/// If a file does not exist then function downloads existing mem-proof for given agent
+/// from HBS server and saves it to the file
+/// Returns error if no memproof obtained, because memproof is mandatory
+/// for core-app installation
+#[instrument(skip(admin), err)]
+pub async fn get_mem_proof(admin: Admin) -> Result<MembraneProof> {
+    if let Ok(m) = load_mem_proof_from_file() {
```

Comment:
As I understand the code, in the case where we:
- Have an existing valid memproof file for the environment (UID + network)
- Delete the `agent_pubkey` file to generate a new random agent

This code will load an invalid memproof file right? (i.e. invalid agent) because it doesn't ensure that the memproof is actually for that agent.

The whole point of having random agents in dev environment is so we can generate new agent_keys. Should we fix it, or should we make it a matter of process that we always delete the entire `/var/lib/configure-holochain` folder instead of only that file?

Reply:
Yeah good point. I was actually thinking that each time script generates [creates agent key in Lair](https://github.com/Holo-Host/hpos-configure-holochain/blob/memproof_logic_simplified/src/agent.rs#L84) it should also delete existing memproof file. That way in next step memproof will be obtained for a correct agent.

This feels much more bullet-proof than requiring devs to always delete entire `/var/lib/configure-holochain`

Reply:
addressed in [982b4e7](https://github.com/Holo-Host/hpos-configure-holochain/pull/143/commits/982b4e70a41e71124209de89c93ca8fef8311e58)